dub.sdl

```diff
@@ -50,17 +51,9 @@ subPackage {
     "CallbackAPI" \
     "DMDLIB"
 
-  preGenerateCommands `
-    "$${DUB_EXE}" \
-    --arch=$${DUB_ARCH} \
-    --compiler=$${DC} \
-    --single "$${DUB_PACKAGE_DIR}config.d" \
-    -- "$${DUB_PACKAGE_DIR}generated/dub" \
-    "$${DUB_PACKAGE_DIR}VERSION" \
-    /etc
-   ` platform="posix"
-
-  preGenerateCommands `"%DUB_EXE%" --arch=%DUB_ARCH% --compiler="%DC%" --single "%DUB_PACKAGE_DIR%config.d" -- "%DUB_PACKAGE_DIR%generated/dub" "%DUB_PACKAGE_DIR%VERSION"` platform="windows"
+  # generate files `generated/dub/{VERSION,SYSCONFDIR.imp}` for string-imports
+  # by building & running the config.d tool
+  preGenerateCommands `"$DUB_EXE" "--compiler=$DC" --single "${DUB_PACKAGE_DIR}config.d" -- "${DUB_PACKAGE_DIR}generated/dub" "${DUB_PACKAGE_DIR}VERSION" /etc`
```

Comment:
The `platform` constraint only applies to the **target** platform, so might be a non-native platform. There's no way to differentiate the **host** platforms.

Reply:
And we generate the `generated/dub/SYSCONFDIR.imp` file on all hosts too now (with content `/etc`). In case the `dmd:lexer` subpackage is ever built for some Posix **target**, it must be string-importable, on Windows **hosts** too: https://github.com/dlang/dmd/blob/83adb8aeab12ef7902eefae81223fdd300cfc1e0/compiler/src/dmd/dinifile.d#L110